m4a-dec.c

```diff
@@ -121,19 +120,24 @@ int main(int argc, char *argv[]) {
 			}
 			frame_size = info->frameSize * info->numChannels;
 			// Note, this probably doesn't return channels > 2 in the right order for wav
-			wav = wav_write_open(outfile, info->sampleRate, 16, info->numChannels);
+			wav = wav_write_open(outfile, info->sampleRate, sizeof(INT_PCM)*8, info->numChannels);
 			if (!wav) {
 				perror(outfile);
 				status = 1;
 				break;
 			}
 		}
-		for (i = 0; i < frame_size; i++) {
-			uint8_t* out = &output_buf[2*i];
+		for (int i = 0; i < frame_size; i++) {
+			uint8_t* out = &output_buf[sizeof(INT_PCM)*i];
 			out[0] = decode_buf[i] & 0xff;
-			out[1] = decode_buf[i] >> 8;
+			out[1] = (uint8_t)(decode_buf[i] >> 8);
+			if(sizeof(INT_PCM) == 2) {
```

Comment:
I'd prefer a space between `if` and `(`, but if you generalize it to a loop for any size, this particular if statement shouldn't be needed